packages/jupyter-widgets/__tests__/manager/widgetLoader.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { requireLoader } from "../../src/manager/widgetLoader";
+
+// A mock valid module representing a custom widget
+const mockModule = {
+  "foo" : "bar"
+};
+// Info representing an invalid module for testing the failure case
+const invalidModule = { 
+  name: "invalid_module",
+  version: "1.0",
+  url: "https://unpkg.com/invalid_module@1.0/dist/index.js"
+};
+// Mock implementation of the core require API
+const mockRequireJS = jest.fn((modules, ready, errCB) => {
+  if(modules.length > 0 && modules[0] == invalidModule.url){
+    errCB(new Error('Whoops!'));
+  }
+  else {
+    ready(mockModule);
+  } 
+});
+
+// Callback binding
+const mockSuccessCB = jest.fn();
+const mockErrorCB = jest.fn();
+(window as any).requirejs = mockRequireJS;
+(window as any).requirejs.config  = jest.fn();
+
+describe("requireLoader", () => {
+  beforeAll(() => {
+    jest.clearAllMocks();
```

Comment:
same comment, is this needed?

Reply:
Yes, I think we need it. Since we have strict assertions on the number of times a mock function is called, we clear all mocks which essentially clears the transient state of a mock function like its call args and other meta data so that we start clean after every test.